pyproject.toml

```diff
@@ -1,10 +1,83 @@
 [build-system]
-build-backend = "setuptools.build_meta"
-requires = [
-    "setuptools>=45.0",
-    "wheel>=0.37.0",
+requires = ["poetry-core>=1.9.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+# Check https://python-poetry.org/docs/pyproject/ for all available sections
+name = "ansys-fluent-core"
+version = "0.20.dev1"
+description = "PyFluent provides Pythonic access to Ansys Fluent"
+license = "MIT"
+authors = ["ANSYS, Inc. <ansys.support@ansys.com>"]
+maintainers = ["PyAnsys developers <pyansys.maintainers@ansys.com>"]
+readme = "README.rst"
+repository = "https://github.com/ansys/pyfluent"
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+]
+
+packages = [
+    { include = "ansys", from = "src" },
+    { include = "tests", to = "ansys/fluent" },
+    { include = "README.rst", to = "ansys/fluent/core/docs" },
 ]
 
+[tool.poetry.dependencies]
+python = ">=3.9,<4.0"
+importlib-metadata = {version = "^7.0.1", python = "<3.9"}
+ansys-api-fluent = "^0.3.22"
+ansys-platform-instancemanagement = "^1.0"
+ansys-units = "^0.3.1"
+beartype = "^0.17"
+docker = "^6.1.3"
+grpcio = "^1.30.0"
+grpcio-health-checking = "^1.30.0"
+lxml = "^4.9.2"
+numpy= "^1.21.5"
+pandas = "^1.1.5"
+platformdirs = "^3.5.1"
+psutil = "^5.9.5"
+pyyaml = "^6.0"
+requests = "^2.31.0"
+
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+Sphinx = "==7.2.6"
+jupyter_sphinx = "==0.5.3"
+numpydoc = "==1.6.0"
+matplotlib = "==3.8.2"
+ansys-sphinx-theme = "==0.13.2"
+pypandoc = "==1.12"
+pytest-sphinx = "==0.5.0"
+sphinx-autobuild = "==2021.3.14"
+sphinx-autodoc-typehints = "==1.25.3"
+sphinx-copybutton = "==0.5.2"
+sphinx-gallery = "==0.15.0"
+sphinx-notfound-page = "==1.0.0"
+sphinxcontrib-websupport = "==1.2.7"
+sphinxemoji = "==0.3.1"
+sphinx-toggleprompt = "==0.5.2"
+autodocsumm = "==0.2.12"
+
+[tool.poetry.group.test]
+optional = true
+
+[tool.poetry.group.test.dependencies]
+pytest-mock = "==3.12.0"
+pytest-cov = "==4.1.0"
+pytest = "==8.0.2"
+h5py = "==3.10.0"
+reader = "==3.7"
```

Comment:
what is this package?

Reply:
It's reader package that needs h5py. 

@mkundu1 I think we can specify it follows.

[tool.poetry.dependencies]
h5py = { version = "^3.10.0", optional = true }

[tool.poetry.extras]
reader= ["h5py "]

Reply:
Done. Thank you.